python/spglib/spglib.py

```diff
@@ -1215,22 +1222,70 @@ def get_magnetic_spacegroup_type(uni_number) -> dict | None:
     """
     _set_no_error()
 
-    keys = (
-        "uni_number",
-        "litvin_number",
-        "bns_number",
-        "og_number",
-        "number",
-        "type",
-    )
     msg_type_list = _spglib.magnetic_spacegroup_type(uni_number)
     _set_error_message()
 
     if msg_type_list is not None:
-        msg_type = dict(zip(keys, msg_type_list))
-        for key in msg_type:
-            if key not in ["uni_number", "litvin_number", "number", "type"]:
-                msg_type[key] = msg_type[key].strip()
+        msg_type = MagneticSpaceGroupType(
+            uni_number=msg_type_list[0],
+            litvin_number=msg_type_list[1],
+            bns_number=msg_type_list[2].strip(),
+            og_number=msg_type_list[3].strip(),
+            number=msg_type_list[4],
+            type=msg_type_list[5],
+        )
+        return msg_type
+    else:
+        return None
+
+
+def get_magnetic_spacegroup_type_from_symmetry(
+    rotations: np.ndarray,
+    translations: np.ndarray,
+    time_reversals: np.ndarray,
+    lattice: np.ndarray | None = None,
+    symprec: float = 1e-5,
+) -> MagneticSpaceGroupType | None:
```

Comment:
A discussion for later, we should really simplify these interface to pass only 1 object